src/Compilers/CSharp/Portable/Symbols/Wrapped/WrappedLocalSymbol.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    /// <summary>
+    /// Represents a local that is based on another local.
+    /// When inheriting from this class, one shouldn't assume that 
+    /// the default behavior it has is appropriate for every case.
+    /// That behavior should be carefully reviewed and derived type
+    /// should override behavior as appropriate.
+    /// </summary>
+    internal abstract class WrappedLocalSymbol : LocalSymbol
+    {
+        public abstract LocalSymbol UnderlyingLocal { get; }
+        public override Symbol ContainingSymbol => UnderlyingLocal.ContainingSymbol;
+        public override TypeWithAnnotations TypeWithAnnotations => UnderlyingLocal.TypeWithAnnotations;
+        public override RefKind RefKind => UnderlyingLocal.RefKind;
+        public override ImmutableArray<Location> Locations => UnderlyingLocal.Locations;
+        public override ImmutableArray<SyntaxReference> DeclaringSyntaxReferences => UnderlyingLocal.DeclaringSyntaxReferences;
+        public override string Name => UnderlyingLocal.Name;
+        public override bool IsImplicitlyDeclared => UnderlyingLocal.IsImplicitlyDeclared;
+        protected override Symbol OriginalSymbolDefinition => (object)UnderlyingLocal.OriginalDefinition == UnderlyingLocal ? this : UnderlyingLocal.OriginalDefinition;
+        internal override LocalDeclarationKind DeclarationKind => UnderlyingLocal.DeclarationKind;
+        internal override SynthesizedLocalKind SynthesizedKind => UnderlyingLocal.SynthesizedKind;
+        internal override SyntaxNode ScopeDesignatorOpt => UnderlyingLocal.ScopeDesignatorOpt;
+        internal override bool IsImportedFromMetadata => UnderlyingLocal.IsImportedFromMetadata;
+        internal override SyntaxToken IdentifierToken => UnderlyingLocal.IdentifierToken;
+        internal override bool IsPinned => UnderlyingLocal.IsPinned;
+        internal override bool IsCompilerGenerated => UnderlyingLocal.IsCompilerGenerated;
+        internal override uint RefEscapeScope => UnderlyingLocal.RefEscapeScope;
+        internal override uint ValEscapeScope => UnderlyingLocal.ValEscapeScope;
+        internal override ConstantValue GetConstantValue(SyntaxNode node, LocalSymbol inProgress, DiagnosticBag diagnostics = null) =>
+            UnderlyingLocal.GetConstantValue(node, inProgress, diagnostics);
+        internal override ImmutableArray<Diagnostic> GetConstantValueDiagnostics(BoundExpression boundInitValue) =>
+            UnderlyingLocal.GetConstantValueDiagnostics(boundInitValue);
+        internal override SyntaxNode GetDeclaratorSyntax() =>
+            UnderlyingLocal.GetDeclaratorSyntax();
+        internal override LocalSymbol WithSynthesizedLocalKindAndSyntax(SynthesizedLocalKind kind, SyntaxNode syntax) =>
```

Comment:
>override [](start = 17, length = 8)

It feels like this API should be abstract because we would want to return an instance of a ```WrappedLocalSymbol```

Reply:
I actually debated making this throw completely. We should never be attempting to do this with a synthesized local.

---
In reply to: [329715544](https://github.com/dotnet/roslyn/pull/38769#discussion_r329715544) [](ancestors = 329715544)

Reply:
Proceeding with making this throw.

---
In reply to: [329828692](https://github.com/dotnet/roslyn/pull/38769#discussion_r329828692) [](ancestors = 329828692,329715544)